docs/DeveloperDocumentation.rst

```diff
@@ -0,0 +1,104 @@
+
+* Creating "settings.xml" file.
+   * Create a file named "settings.xml" inside ``.m2`` folder. It is a hidden folder. To view it go to your home directory press ``command + shift + .`` (if you are on a Mac, if you are on a different OS, show all hidden files using your OS specific method). All hidden files should show up.
+   * Now open .m2 folder and create a file ``settings.xml``.
+   * Create a folder uploadFiles and keep note of its path.
+   * Copy below code into it.
+
+.. code-block:: xml
+   
+      <settings>
+         <profiles>
+            <profile>
+               <id>citesphere</id>
+               <properties>
+                  <zotero.client.key>your_zotero_key</zotero.client.key>
+                  <zotero.client.secret>your_zotero_secret</zotero.client.secret>
+                  <upload.folder.path>/Users/your_path/uploadFiles</upload.folder.path>
+                  <token.signing.key>EsAxEfq/AGPGfPEJ/xEVSjVAU7H8GThexbfNnsrmaRYVREoxAJJEXDPKDO0GE87vhL1Z3OJz88CACq
+                  C4lKJ5TA==</token.signing.key>
+                  <db.database.url>jdbc:mysql://localhost:3306/citesphere</db.database.url>
+                  <db.user>mysql_db_username</db.user>
+                  <db.password>mysql_password</db.password>
+                  <app.url>http://localhost:8080/citesphere</app.url>
+                  <kafka.hosts>localhost:9092</kafka.hosts>
+                  <db.hibernate.dialect>org.hibernate.dialect.MySQL5InnoDBDialect</db.hibernate.dialect>
+               </properties>
+            </profile>
+         </profiles>
+      </settings>
+      
+in the above xml the variables need to be adjusted to the specific system settings (e.g. zotero.client.key needs to be replaced with the client key created earlier in Zotero, MySLQ username and password need to be replaced, etc).
+
+* Start MySQL
+   * Open terminal enter this command ``/usr/local/mysql/bin/mysql -u your_mysql_db_username -p``. 
+     If you don't have any additional user other than root just give root as username. Press enter. Give your MySQL password.
+      
+   * create another user. Use this command. ``CREATE USER 'user_name'@'localhost' IDENTIFIED BY 'password';``
+   
+   * Grant all privileges.  ``GRANT ALL PRIVILEGES ON * . * TO 'user_name'@'localhost';``.
+   
+   * update new username and password in ``settings.xml`` file which is inside .m2 folder.
+   
+   * Type ``exit`` you will come out of MySQL terminal.
+   
+   * Now connect to MySQL with new username and password you just created ``/usr/local/mysql/bin/mysql -u your_mysql_db_username -p``.
+   
+
+* Start Apache Kafka and Zookeeper.
+   * To start Zookeeper open terminal and use this command. ``sudo /Users/your_path/zookeeper-3.3.6/bin/zkServer.sh start``. 
+   
+   * To start Kafka use this ``sudo /Users/your_path/kafka_2.11-1.1.1/bin/kafka-server-start.sh /Users/your_path/kafka_2.11   1.1.1/config/server.properties``  (there is space in between start.sh and /Users)
```

Comment:
there seems to be an issue with the kafka folder name (why are there several spaces between `2.11` and `1.1.1`. Also, the same version placeholder as above should be used.

Reply:
and `start.sh` and `/Users` should be formatted as code.